vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -1,1011 +1,508 @@
+                        result.id.counterId,
                     )
                     continue
 
-                for result in mor_perfs.value:
-                    counter_id = result.id.counterId
-                    if not self.metadata_cache.contains(i_key, counter_id):
-                        self.log.debug(
-                            "Skipping value for counter %s, because there is no metadata about it",
-                            ensure_unicode(counter_id),
-                        )
-                        continue
-
-                    # Metric types are absolute, delta, and rate
-                    metric_name = self.metadata_cache.get_metadata(i_key, result.id.counterId).get('name')
-
-                    if self.in_compatibility_mode(instance):
-                        if metric_name not in ALL_METRICS:
-                            self.log.debug("Skipping unknown `%s` metric.", ensure_unicode(metric_name))
-                            continue
-
-                    if not result.value:
-                        self.log.debug("Skipping `%s` metric because the value is empty", ensure_unicode(metric_name))
-                        continue
-
-                    instance_name = result.id.instance or "none"
-                    # Get the most recent value that isn't negative
-                    valid_values = [v for v in result.value if v >= 0]
-                    if not valid_values:
-                        continue
-                    value = self._transform_value(instance, result.id.counterId, valid_values[-1])
-
-                    hostname = mor['hostname']
-
-                    tags = ['instance:{}'.format(ensure_unicode(instance_name))]
-                    if not hostname:  # no host tags available
-                        tags.extend(mor['tags'])
-                    else:
-                        hostname = to_string(hostname)
-                        if self.excluded_host_tags:
-                            tags.extend(mor["excluded_host_tags"])
+                if not result.value:
+                    self.log.debug("Skipping metric %s because the value is empty", ensure_unicode(metric_name))
+                    continue
 
-                    tags.extend(instance.get('tags', []))
+                # Get the most recent value that isn't negative
+                valid_values = [v for v in result.value if v >= 0]
+                if not valid_values:
+                    self.log.debug(
+                        "Skipping metric %s because the value returned by vCenter"
+                        " is negative (i.e. the metric is not yet available).",
+                        ensure_unicode(metric_name),
+                    )
+                    continue
+                value = valid_values[-1]
+                if metric_name in PERCENT_METRICS:
+                    # Convert the percentage to a float.
+                    value = float(value) / 100
```

Comment:
If you put back `from __future__ import division`

```suggestion
                    value /= 100
```

Reply:
done